h/static/scripts/test/streamer-test.js

```diff
@@ -83,27 +84,66 @@ describe('streamer', function () {
   it('should not create a websocket connection if websocketUrl is not provided', function () {
     fakeSettings = {};
     createDefaultStreamer();
+    activeStreamer.connect();
     assert.isNull(fakeWebSocket);
   });
 
-  it('should send a client ID', function () {
+  it('should not create a websocket connection if the user is not logged in', function () {
+    createDefaultStreamer();
+    assert.isNull(fakeWebSocket);
+  });
+
+  it('should create a websocket connection if the user is logged in', function () {
+    fakeSession.state = { userid: "foo" };
+    createDefaultStreamer();
+    assert.ok(fakeWebSocket);
+  });
+
+  it('should create a websocket connection if explicitly connected', function () {
+    createDefaultStreamer();
+    activeStreamer.connect();
+    assert.ok(fakeWebSocket);
+  });
+
+  it('should have a non-null client ID', function () {
     createDefaultStreamer();
+    assert.ok(activeStreamer.clientId);
+  });
+
+  it('should send the client ID on connection', function () {
+    createDefaultStreamer();
+    activeStreamer.connect();
     assert.equal(fakeWebSocket.messages.length, 1);
     assert.equal(fakeWebSocket.messages[0].messageType, 'client_id');
     assert.equal(fakeWebSocket.messages[0].value, activeStreamer.clientId);
   });
 
-  it('should close any existing socket', function () {
-    createDefaultStreamer();
-    var oldWebSocket = fakeWebSocket;
-    activeStreamer.connect();
-    assert.ok(oldWebSocket.didClose);
-    assert.ok(!fakeWebSocket.didClose);
+  context('.connect()', function () {
```

Comment:
Using JSDoc conventions, tests for instance methods are usually written as `describe('#methodName()')` whereas tests for class/static methods would be written as `describe('.foo()'`)